Type 'Entity' does not satisfy the constraint 'ObjectLiteral'. update(): UpdateQueryBuilder<Entity>;

<!--
  Please follow the template.  If you don't, your issue may be closed.

  Have a question?  This is the TypeORM issue tracker - and not the right place
  for general support or questions.  Instead, check the "Support" Documentation
  on the best places to ask questions!

  https://github.com/typeorm/typeorm/blob/master/docs/support.md
-->


### Expected Behavior
Just Build without any error.

## Issue Description
 Build Time error.
 It Work perfectly in development. When run build (tsc -p .)  then getting this error below.

<!--
  A clear and concise description of what actually happened.

  Please wrap any error messages or output in code tags, instead of images.
-->

```
node_modules/typeorm/query-builder/QueryBuilder.d.ts:84:34 - error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.

84     update(): UpdateQueryBuilder<Entity>;
                                    ~~~~~~

  node_modules/typeorm/query-builder/QueryBuilder.d.ts:24:44
    24 export declare abstract class QueryBuilder<Entity> {
                                                  ~~~~~~
    This type parameter might need an `extends ObjectLiteral` constraint.

node_modules/typeorm/query-builder/QueryBuilder.d.ts:88:75 - error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.

88     update(updateSet: QueryDeepPartialEntity<Entity>): UpdateQueryBuilder<Entity>;
                                                                             ~~~~~~

  node_modules/typeorm/query-builder/QueryBuilder.d.ts:24:44
    24 export declare abstract class QueryBuilder<Entity> {
                                                  ~~~~~~
    This type parameter might need an `extends ObjectLiteral` constraint.

node_modules/typeorm/query-builder/QueryBuilder.d.ts:92:114 - error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.

92     update<Entity>(entity: EntityTarget<Entity>, updateSet?: QueryDeepPartialEntity<Entity>): UpdateQueryBuilder<Entity>;      
                                                                                                                    ~~~~~~        

  node_modules/typeorm/query-builder/QueryBuilder.d.ts:92:12
    92     update<Entity>(entity: EntityTarget<Entity>, updateSet?: QueryDeepPartialEntity<Entity>): UpdateQueryBuilder<Entity>;  
                  ~~~~~~
    This type parameter might need an `extends ObjectLiteral` constraint.

node_modules/typeorm/query-builder/QueryBuilder.d.ts:96:95 - error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.

96     update(tableName: string, updateSet?: QueryDeepPartialEntity<Entity>): UpdateQueryBuilder<Entity>;
                                                                                                 ~~~~~~

  node_modules/typeorm/query-builder/QueryBuilder.d.ts:24:44
    24 export declare abstract class QueryBuilder<Entity> {
                                                  ~~~~~~
    This type parameter might need an `extends ObjectLiteral` constraint.


Found 4 errors in the same file, starting at: node_modules/typeorm/query-builder/QueryBuilder.d.ts:84
```


### Steps to Reproduce

<!--
  Your bug will be investigated much faster if we can run your code and it doesn't
  have dependencies other than TypeORM. Issues without reproduction steps or
  code examples may be closed as not actionable.

  Please try to provide a Minimal, Complete, and Verifiable example.
  http://stackoverflow.com/help/mcve
-->

### My Environment

<!--
  Please add any other relevant dependencies to this table at the end.
  For example: Electron, React Native, or NestJS.
-->

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| Node.js version     | v16.15.0  | <!-- run `node -v` to obtain this -->
| Typescript version  | 4.8.2  | <!-- run `npm list typescript` to obtain this -->
| TypeORM version     | 0.3.8  | <!-- run `npm list typeorm` to obtain this -->
| Express     | 4.18.1  | <!-- run `npm list typeorm` to obtain this -->


### Additional Context

<!--
  Add any other context about the bug report here.
-->


### Relevant Database Driver(s)

<!-- Put "yes" instead of "no" to your issue's relevant databases -->

| DB Type           | Reproducible  |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | yes|
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | no            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | no            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |


<!-- Put "✅" (:white_check_mark:) to one of these options, left "✖️" (:heavy_multiplication_x:) others: -->

 - ✖️ Yes, I have the time, and I know how to start.
 - ✅ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.


<!--
  Remember that first-time contributors are welcome! 🙌
  👋 Have a great day and thank you for the bug report!
-->

We are using version `0.2.45` and are faced with the same issue. It is difficult for us to migrate to typeorm 0.3 at this time and we would like to use the latest TS version. Is it possible to come up with a patch for typeorm 0.2? Here is the error we are seeing:
```
$ yarn clean
$ rimraf ./dist
$ tsc
../../node_modules/typeorm/connection/Connection.d.ts(150,69): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/connection/Connection.d.ts(160,79): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/entity-manager/EntityManager.d.ts(318,69): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/entity-manager/EntityManager.d.ts(329,79): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/globals.d.ts(71,119): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/globals.d.ts(83,129): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/repository/AbstractRepository.d.ts(41,70): error TS2344: Type 'T' does not satisfy the constraint 'ObjectLiteral'.
../../node_modules/typeorm/repository/TreeRepository.d.ts(10,64): error TS2344: Type 'Entity' does not satisfy the constraint 'ObjectLiteral'.
```

--- a/package.json
+++ b/package.json
@@ -141,7 +141,7 @@
     "sqlite3": "^5.0.11",
     "ts-node": "^10.7.0",
     "typeorm-aurora-data-api-driver": "^2.0.0",
-    "typescript": "^4.6.2"
+    "typescript": "^4.8.2"
   },
   "peerDependencies": {
     "@google-cloud/spanner": "^5.18.0",

--- a/src/data-source/DataSource.ts
+++ b/src/data-source/DataSource.ts
@@ -521,7 +521,7 @@ export class DataSource {
     /**
      * Creates a new query builder that can be used to build a SQL query.
      */
-    createQueryBuilder<Entity>(
+    createQueryBuilder<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         alias: string,
         queryRunner?: QueryRunner,
@@ -535,7 +535,7 @@ export class DataSource {
     /**
      * Creates a new query builder that can be used to build a SQL query.
      */
-    createQueryBuilder<Entity>(
+    createQueryBuilder<Entity extends ObjectLiteral>(
         entityOrRunner?: EntityTarget<Entity> | QueryRunner,
         alias?: string,
         queryRunner?: QueryRunner,

--- a/src/entity-manager/EntityManager.ts
+++ b/src/entity-manager/EntityManager.ts
@@ -175,7 +175,7 @@ export class EntityManager {
     /**
      * Creates a new query builder that can be used to build a SQL query.
      */
-    createQueryBuilder<Entity>(
+    createQueryBuilder<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         alias: string,
         queryRunner?: QueryRunner,
@@ -189,7 +189,7 @@ export class EntityManager {
     /**
      * Creates a new query builder that can be used to build a SQL query.
      */
-    createQueryBuilder<Entity>(
+    createQueryBuilder<Entity extends ObjectLiteral>(
         entityClass?: EntityTarget<Entity> | QueryRunner,
         alias?: string,
         queryRunner?: QueryRunner,
@@ -659,7 +659,7 @@ export class EntityManager {
      * Does not check if entity exist in the database, so query will fail if duplicate entity is being inserted.
      * You can execute bulk inserts using this method.
      */
-    async insert<Entity>(
+    async insert<Entity extends ObjectLiteral>(
         target: EntityTarget<Entity>,
         entity:
             | QueryDeepPartialEntity<Entity>
@@ -672,7 +672,7 @@ export class EntityManager {
             .execute()
     }
 
-    async upsert<Entity>(
+    async upsert<Entity extends ObjectLiteral>(
         target: EntityTarget<Entity>,
         entityOrEntities:
             | QueryDeepPartialEntity<Entity>
@@ -736,7 +736,7 @@ export class EntityManager {
      * Does not check if entity exist in the database.
      * Condition(s) cannot be empty.
      */
-    update<Entity>(
+    update<Entity extends ObjectLiteral>(
         target: EntityTarget<Entity>,
         criteria:
             | string
@@ -791,7 +791,7 @@ export class EntityManager {
      * Does not check if entity exist in the database.
      * Condition(s) cannot be empty.
      */
-    delete<Entity>(
+    delete<Entity extends ObjectLiteral>(
         targetOrEntity: EntityTarget<Entity>,
         criteria:
             | string
@@ -950,7 +950,7 @@ export class EntityManager {
      * Counts entities that match given options.
      * Useful for pagination.
      */
-    count<Entity>(
+    count<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         options?: FindManyOptions<Entity>,
     ): Promise<number> {
@@ -968,7 +968,7 @@ export class EntityManager {
      * Counts entities that match given conditions.
      * Useful for pagination.
      */
-    countBy<Entity>(
+    countBy<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         where: FindOptionsWhere<Entity> | FindOptionsWhere<Entity>[],
     ): Promise<number> {
@@ -981,7 +981,7 @@ export class EntityManager {
     /**
      * Finds entities that match given find options.
      */
-    async find<Entity>(
+    async find<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         options?: FindManyOptions<Entity>,
     ): Promise<Entity[]> {
@@ -998,7 +998,7 @@ export class EntityManager {
     /**
      * Finds entities that match given find options.
      */
-    async findBy<Entity>(
+    async findBy<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         where: FindOptionsWhere<Entity> | FindOptionsWhere<Entity>[],
     ): Promise<Entity[]> {
@@ -1016,7 +1016,7 @@ export class EntityManager {
      * Also counts all entities that match given conditions,
      * but ignores pagination settings (from and take options).
      */
-    findAndCount<Entity>(
+    findAndCount<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         options?: FindManyOptions<Entity>,
     ): Promise<[Entity[], number]> {
@@ -1035,7 +1035,7 @@ export class EntityManager {
      * Also counts all entities that match given conditions,
      * but ignores pagination settings (from and take options).
      */
-    findAndCountBy<Entity>(
+    findAndCountBy<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         where: FindOptionsWhere<Entity> | FindOptionsWhere<Entity>[],
     ): Promise<[Entity[], number]> {
@@ -1058,7 +1058,7 @@ export class EntityManager {
      *     id: In([1, 2, 3])
      * })
      */
-    async findByIds<Entity>(
+    async findByIds<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         ids: any[],
     ): Promise<Entity[]> {
@@ -1078,7 +1078,7 @@ export class EntityManager {
      * Finds first entity by a given find options.
      * If entity was not found in the database - returns null.
      */
-    async findOne<Entity>(
+    async findOne<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         options: FindOneOptions<Entity>,
     ): Promise<Entity | null> {
@@ -1109,7 +1109,7 @@ export class EntityManager {
      * Finds first entity that matches given where condition.
      * If entity was not found in the database - returns null.
      */
-    async findOneBy<Entity>(
+    async findOneBy<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         where: FindOptionsWhere<Entity> | FindOptionsWhere<Entity>[],
     ): Promise<Entity | null> {
@@ -1134,7 +1134,7 @@ export class EntityManager {
      *     id: 1 // where "id" is your primary column name
      * })
      */
-    async findOneById<Entity>(
+    async findOneById<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         id: number | string | Date | ObjectID,
     ): Promise<Entity | null> {
@@ -1153,7 +1153,7 @@ export class EntityManager {
      * Finds first entity by a given find options.
      * If entity was not found in the database - rejects with error.
      */
-    async findOneOrFail<Entity>(
+    async findOneOrFail<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         options: FindOneOptions<Entity>,
     ): Promise<Entity> {
@@ -1173,7 +1173,7 @@ export class EntityManager {
      * Finds first entity that matches given where condition.
      * If entity was not found in the database - rejects with error.
      */
-    async findOneByOrFail<Entity>(
+    async findOneByOrFail<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         where: FindOptionsWhere<Entity> | FindOptionsWhere<Entity>[],
     ): Promise<Entity> {
@@ -1209,7 +1209,7 @@ export class EntityManager {
     /**
      * Increments some column by provided value of the entities matched given conditions.
      */
-    async increment<Entity>(
+    async increment<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         conditions: any,
         propertyPath: string,
@@ -1246,7 +1246,7 @@ export class EntityManager {
     /**
      * Decrements some column by provided value of the entities matched given conditions.
      */
-    async decrement<Entity>(
+    async decrement<Entity extends ObjectLiteral>(
         entityClass: EntityTarget<Entity>,
         conditions: any,
         propertyPath: string,
@@ -1354,8 +1354,8 @@ export class EntityManager {
      * sets current EntityManager instance to it. Used to work with custom repositories
      * in transactions.
      */
-    withRepository<Entity extends ObjectLiteral, R extends Repository<Entity>>(
-        repository: R,
+    withRepository<Entity extends ObjectLiteral, R extends Repository<any>>(
+        repository: R & Repository<Entity>,
     ): R {
         const repositoryConstructor =
             repository.constructor as typeof Repository

--- a/src/find-options/FindOptionsUtils.ts
+++ b/src/find-options/FindOptionsUtils.ts
@@ -5,6 +5,7 @@ import { FindRelationsNotFoundError } from "../error/FindRelationsNotFoundError"
 import { EntityMetadata } from "../metadata/EntityMetadata"
 import { DriverUtils } from "../driver/DriverUtils"
 import { FindTreeOptions } from "./FindTreeOptions"
+import { ObjectLiteral } from "../common/ObjectLiteral"
 
 /**
  * Utilities to work with FindOptions.
@@ -239,7 +240,7 @@ export class FindOptionsUtils {
         return qb;
     }*/
 
-    static applyOptionsToTreeQueryBuilder<T>(
+    static applyOptionsToTreeQueryBuilder<T extends ObjectLiteral>(
         qb: SelectQueryBuilder<T>,
         options?: FindTreeOptions,
     ): SelectQueryBuilder<T> {

--- a/src/query-builder/DeleteQueryBuilder.ts
+++ b/src/query-builder/DeleteQueryBuilder.ts
@@ -12,7 +12,7 @@ import { InstanceChecker } from "../util/InstanceChecker"
 /**
  * Allows to build complex sql queries in a fashion way and execute those queries.
  */
-export class DeleteQueryBuilder<Entity>
+export class DeleteQueryBuilder<Entity extends ObjectLiteral>
     extends QueryBuilder<Entity>
     implements WhereExpressionBuilder
 {
@@ -116,7 +116,7 @@ export class DeleteQueryBuilder<Entity>
      * Specifies FROM which entity's table select/update/delete will be executed.
      * Also sets a main string alias of the selection data.
      */
-    from<T>(
+    from<T extends ObjectLiteral>(
         entityTarget: EntityTarget<T>,
         aliasName?: string,
     ): DeleteQueryBuilder<T> {

--- a/src/query-builder/InsertQueryBuilder.ts
+++ b/src/query-builder/InsertQueryBuilder.ts
@@ -21,7 +21,9 @@ import { InstanceChecker } from "../util/InstanceChecker"
 /**
  * Allows to build complex sql queries in a fashion way and execute those queries.
  */
-export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
+export class InsertQueryBuilder<
+    Entity extends ObjectLiteral,
+> extends QueryBuilder<Entity> {
     readonly "@instanceof" = Symbol.for("InsertQueryBuilder")
 
     // -------------------------------------------------------------------------
@@ -227,7 +229,7 @@ export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
     /**
      * Specifies INTO which entity's table insertion will be executed.
      */
-    into<T>(
+    into<T extends ObjectLiteral>(
         entityTarget: EntityTarget<T>,
         columns?: string[],
     ): InsertQueryBuilder<T> {

--- a/src/query-builder/QueryBuilder.ts
+++ b/src/query-builder/QueryBuilder.ts
@@ -44,7 +44,7 @@ import { escapeRegExp } from "../util/escapeRegExp"
 /**
  * Allows to build complex sql queries in a fashion way and execute those queries.
  */
-export abstract class QueryBuilder<Entity> {
+export abstract class QueryBuilder<Entity extends ObjectLiteral> {
     readonly "@instanceof" = Symbol.for("QueryBuilder")
 
     // -------------------------------------------------------------------------
@@ -216,7 +216,7 @@ export abstract class QueryBuilder<Entity> {
     /**
      * Creates UPDATE query for the given entity and applies given update values.
      */
-    update<Entity>(
+    update<Entity extends ObjectLiteral>(
         entity: EntityTarget<Entity>,
         updateSet?: QueryDeepPartialEntity<Entity>,
     ): UpdateQueryBuilder<Entity>
@@ -308,7 +308,7 @@ export abstract class QueryBuilder<Entity> {
     /**
      * Sets entity's relation with which this query builder gonna work.
      */
-    relation<T>(
+    relation<T extends ObjectLiteral>(
         entityTarget: EntityTarget<T>,
         propertyPath: string,
     ): RelationQueryBuilder<T>

--- a/src/query-builder/QueryPartialEntity.ts
+++ b/src/query-builder/QueryPartialEntity.ts
@@ -1,3 +1,5 @@
+import { ObjectLiteral } from "../common/ObjectLiteral"
+
 /**
  * Make all properties in T optional
  */
@@ -8,12 +10,16 @@ export type QueryPartialEntity<T> = {
 /**
  * Make all properties in T optional. Deep version.
  */
-export type QueryDeepPartialEntity<T> = {
+export type QueryDeepPartialEntity<T> = _QueryDeepPartialEntity<
+    ObjectLiteral extends T ? unknown : T
+>
+
+type _QueryDeepPartialEntity<T> = {
     [P in keyof T]?:
         | (T[P] extends Array<infer U>
-              ? Array<QueryDeepPartialEntity<U>>
+              ? Array<_QueryDeepPartialEntity<U>>
               : T[P] extends ReadonlyArray<infer U>
-              ? ReadonlyArray<QueryDeepPartialEntity<U>>
-              : QueryDeepPartialEntity<T[P]>)
+              ? ReadonlyArray<_QueryDeepPartialEntity<U>>
+              : _QueryDeepPartialEntity<T[P]>)
         | (() => string)
 }

--- a/src/query-builder/RelationQueryBuilder.ts
+++ b/src/query-builder/RelationQueryBuilder.ts
@@ -3,13 +3,16 @@ import { RelationUpdater } from "./RelationUpdater"
 import { RelationRemover } from "./RelationRemover"
 import { TypeORMError } from "../error"
 import { ObjectUtils } from "../util/ObjectUtils"
+import { ObjectLiteral } from "../common/ObjectLiteral"
 
 /**
  * Allows to work with entity relations and perform specific operations with those relations.
  *
  * todo: add transactions everywhere
  */
-export class RelationQueryBuilder<Entity> extends QueryBuilder<Entity> {
+export class RelationQueryBuilder<
+    Entity extends ObjectLiteral,
+> extends QueryBuilder<Entity> {
     readonly "@instanceof" = Symbol.for("RelationQueryBuilder")
 
     // -------------------------------------------------------------------------

--- a/src/query-builder/SelectQueryBuilder.ts
+++ b/src/query-builder/SelectQueryBuilder.ts
@@ -47,7 +47,7 @@ import { InstanceChecker } from "../util/InstanceChecker"
 /**
  * Allows to build complex sql queries in a fashion way and execute those queries.
  */
-export class SelectQueryBuilder<Entity>
+export class SelectQueryBuilder<Entity extends ObjectLiteral>
     extends QueryBuilder<Entity>
     implements WhereExpressionBuilder
 {
@@ -254,7 +254,7 @@ export class SelectQueryBuilder<Entity>
      * Also sets a main string alias of the selection data.
      * Removes all previously set from-s.
      */
-    from<T>(
+    from<T extends ObjectLiteral>(
         entityTarget: (qb: SelectQueryBuilder<any>) => SelectQueryBuilder<any>,
         aliasName: string,
     ): SelectQueryBuilder<T>
@@ -264,7 +264,7 @@ export class SelectQueryBuilder<Entity>
      * Also sets a main string alias of the selection data.
      * Removes all previously set from-s.
      */
-    from<T>(
+    from<T extends ObjectLiteral>(
         entityTarget: EntityTarget<T>,
         aliasName: string,
     ): SelectQueryBuilder<T>
@@ -274,7 +274,7 @@ export class SelectQueryBuilder<Entity>
      * Also sets a main string alias of the selection data.
      * Removes all previously set from-s.
      */
-    from<T>(
+    from<T extends ObjectLiteral>(
         entityTarget:
             | EntityTarget<T>
             | ((qb: SelectQueryBuilder<any>) => SelectQueryBuilder<any>),
@@ -289,7 +289,7 @@ export class SelectQueryBuilder<Entity>
      * Specifies FROM which entity's table select/update/delete will be executed.
      * Also sets a main string alias of the selection data.
      */
-    addFrom<T>(
+    addFrom<T extends ObjectLiteral>(
         entityTarget: (qb: SelectQueryBuilder<any>) => SelectQueryBuilder<any>,
         aliasName: string,
     ): SelectQueryBuilder<T>
@@ -298,7 +298,7 @@ export class SelectQueryBuilder<Entity>
      * Specifies FROM which entity's table select/update/delete will be executed.
      * Also sets a main string alias of the selection data.
      */
-    addFrom<T>(
+    addFrom<T extends ObjectLiteral>(
         entityTarget: EntityTarget<T>,
         aliasName: string,
     ): SelectQueryBuilder<T>
@@ -307,7 +307,7 @@ export class SelectQueryBuilder<Entity>
      * Specifies FROM which entity's table select/update/delete will be executed.
      * Also sets a main string alias of the selection data.
      */
-    addFrom<T>(
+    addFrom<T extends ObjectLiteral>(
         entityTarget:
             | EntityTarget<T>
             | ((qb: SelectQueryBuilder<any>) => SelectQueryBuilder<any>),

--- a/src/repository/BaseEntity.ts
+++ b/src/repository/BaseEntity.ts
@@ -216,7 +216,7 @@ export class BaseEntity {
         this: { new (): T } & typeof BaseEntity,
         entityLike: DeepPartial<T>,
     ): Promise<T | undefined> {
-        const thisRepository = this.getRepository() as Repository<T>
+        const thisRepository = this.getRepository<T>()
         return thisRepository.preload(entityLike)
     }
 

